python/cudaq/kernel/ast_bridge.py

```diff
@@ -265,6 +265,43 @@ def visit_FunctionDef(self, node):
 
         ValidateArgumentAnnotations(self).visit(astModule)
 
+        # Ensure that functions with a return-type annotation actually has a valid return statement
+        # in all paths, if not throw an error.
+        class ValidateReturnStatements(ast.NodeVisitor):
```

Comment:
This should not be about requiring a `return` statement in the body of all `if`s in a kernel. For example, the following kernel is valid:

```python
@cudaq.kernel
def kernel(cond :bool) -> int:
    result = 0
    if cond:
        a = 1

    return a
```

This must be about proving whether the end of the kernel body is reachable.

```python
@cudaq.kernel
def kernel(cond :bool) -> int:
    if ... :
        for ... in ... :
            if ... :
               ...
            else:
               ...
    else:
       ...

    # The question is whether this point here is reachable or not.

    # The bridge has to either prove that this is unreachable, and thus
    # it is safe to silently add the `return` (in case the user didn't),
    # or complain to the user about a missing return.
```